accepted/00000-ptf-repos-in-uyuni.md

```diff
@@ -0,0 +1,162 @@
+
+As SCC needs to crawl the buildservice to find all the existing PTF repositories, they agreed on a path structure to map the repositories to the products.
+The defined path is:
+
+    PTF/Release/<Customer Account ID>/<Product Identifier>/<Product Version>/<Product Architecture>/ptf[_debug]/
+
+When Uyuni is listing the repositories available for an organizations, it may find repositories which `id` is not known in the complete product tree.
+To identify them as PTF repositories we can check if the path start with `PTF/Release`.
+
+If they are PTF repositories, we need to generate the missing values for the Channel mapping like this:
+
+- scc repository id: take it from the repository definition
+- scc product id: search for the product using `Product Identifier`, `Product Version`, and `Product Architecture` like specified in the URL
+- scc root product id: find all existing root products for the product id. We need to generate an entry for all of them
+
+The values for update tag and mandatory can be hard coded as they are the same 
+for all PTF repositories.
+
+- update tag: empty [1]
+- mandatory: false
+
+As PTF repositories do not get patches, an update tag is not needed.
+The mandatory flag should be `false` as PTF repositories should only be used when needed.
+
+As we need to generate entries for all existing root products we can iterate over all we found.
+For every root product we take the first channel we can find which does have a `parent channel label` set.
+We can take this as parent channel label for our new entry.
+
+As the channel label and name must be unique.
+We can generate the label and name from the URL components:
+
+    <Customer Account ID> <Product Identifier> <Product Version> "PTFs" ["Debuginfo"] <Product Architecture>
+
+The label is computed using the lowercase components joined with a dash ("-") and the name just joined with a space (" ").
+
+This can either be done by Uyuni Server or SCC. In case SCC use this algorithm it could set it as `repository name`
+joined with a dash ("-") and in `repository description` using spaces.
+
+The product tree uses additional suffixes for products which belongs to multiple root products to make label and name unique.
+The suffix is appended always after the architecture of the channel label.
+We can parse it from the entry we used to find the `parent channel label` and use it to enhance the component list.
+
+
+## Test Package Channels
+
+Before generating PTFs, SUSE will first create so called TEST packages.
+The customer need to test them and report positive result before SUSE make an official supported PTF out of it.
+
+TEST packages should only be installed on dedicated test systems.
+Therefor it is important that they get not installed on any system where just the version of the TEST package is higher than the installed one.
+
+Current idea is, that TEST packages will go into a different channel. Probably the last directory will be names "test" and "test_debug".
```

Comment:
@mcalmer has anything changed on this front lately?

I would ask to remove the "Current idea" language from the RFC, as it should define what is happening/decided/set at this point in time (if at all possible).

Reply:
@moio I can remember there was a talk about specific naming standard of PTF repos to help SUSE Manager. 
@ngetahun @mssola do you know if we have this tracked as feature request?

Reply:
Last time I discussed this with L3 people, we agreed that for now repositories on IBS would be exactly named `ptf`. Other repositories from subprojects within `PTF:Release` would be simply ignored. Thus, for now, they will never be available as a PTF repository or any of that.

When discussing this, it was said that how to handle this scenario was not clear yet from the zypper side of things, and how to adapt all of this with APIs was also to be discussed.

Reply:
I removed the "debug" channels from the RFC. Test repos might come later as this text say now. For now it is just an option with some clarifications who we would handle it.